interfaces/builtin/repowerd.go

```diff
@@ -0,0 +1,335 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var repowerdPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the repowerd service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# HW alarms
+/dev/alarm rw,
+
+# Suspend
+/sys/power/** rw,
+
+# Backlight
+/sys/**/brightness rw,
+/sys/**/max_brightness r,
+/sys/**/type r,
+
+# DBus
+#include <abstractions/dbus-strict>
+
+dbus (send)
+  bus=system
+  path=/org/freedesktop/DBus
+  interface=org.freedesktop.DBus
+  member={GetId,RequestName,ReleaseName,GetConnectionUnixProcessID}
+  peer=(name=org.freedesktop.DBus),
+
+dbus (receive, send)
+  bus=system
+  interface=org.freedesktop.DBus.{Properties,Introspectable},
+
+dbus (bind)
+  bus=system
+  name=com.canonical.powerd,
+
+dbus (bind)
+  bus=system
+  name=com.canonical.Unity.Screen,
+
+dbus (receive, send)
+  bus=system
+  interface=com.canonical.powerd,
+
+dbus (receive, send)
+  bus=system
+  interface=com.canonical.Unity.Screen,
+
+# logind session
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member=GetSessionByPID
+  peer=(name=org.freedesktop.login1),
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1{,/seat/**,/session/**}
+  interface=org.freedesktop.DBus.Properties
+  member=Get{,All}
+  peer=(name=org.freedesktop.login1),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member={SessionAdded,SessionRemoved}
+  peer=(name=org.freedesktop.login1),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/login1{,/seat/**,/session/**}
+  interface=org.freedesktop.DBus.Properties
+  member=PropertiesChanged
+  peer=(name=org.freedesktop.login1),
+
+# logind power
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member={Hibernate,HybridSleep,Inhibit,PowerOff,Suspend}
+  peer=(name=org.freedesktop.login1),
+
+# UPower
```

Comment:
You shouldn't mix plug and slot here. What clients of repowerd only need is a plug of the repowerd interface. They don't need to use the ofono, .. interfaces as this is only a thing the repowerd snap itself needs to use in order to get access to those services. We're using this scheme all over the place already and hardcoding everything into a single interface just makes maintenance pretty hard.